api/src/main/java/io/druid/data/input/impl/StringDimensionSchema.java

```diff
@@ -27,15 +27,19 @@
 {
   @JsonCreator
   public static StringDimensionSchema create(String name) {
-    return new StringDimensionSchema(name);
+    return new StringDimensionSchema(name, -1);
   }
 
+  private final int compareCacheSize;
+
   @JsonCreator
   public StringDimensionSchema(
-      @JsonProperty("name") String name
+      @JsonProperty("name") String name,
+      @JsonProperty("compareCacheSize") int compareCacheSize
```

Comment:
I've also thought of that(where to configure?). but the effectivity of cache is based on the cardinality of the dimension and I've thought it should be configurable per dimension.

Is there a process directly merging two incremental index datum? If it's not like that, it'll not make a problem.